pkg/probe/exec/exec.go

```diff
@@ -39,7 +46,52 @@ type execProber struct{}
 // from executing a command. Returns the Result status, command output, and
 // errors if any.
 func (pr execProber) Probe(e exec.Cmd) (probe.Result, string, error) {
-	data, err := e.CombinedOutput()
+	stdoutPipe, err := e.StdoutPipe()
+	if err != nil {
+		exit, ok := err.(exec.ExitError)
+		if ok && exit.ExitStatus() != 0 {
+			return probe.Failure, "", nil
+		}
+		return probe.Unknown, "", fmt.Errorf("error reading stdout : %s", err)
+	}
+	stderrPipe, err := e.StderrPipe()
+	if err != nil {
+		exit, ok := err.(exec.ExitError)
+		if ok && exit.ExitStatus() != 0 {
+			return probe.Failure, "", nil
+		}
+		return probe.Unknown, "", fmt.Errorf("error reading stderr : %s", err)
+	}
+
+	stdout := make(chan string)
+	stderr := make(chan string)
+
+	go func() {
+		b, _ := utilio.ReadAtMost(stdoutPipe, maxReadLength)
+		scanner := bufio.NewScanner(stdoutPipe)
+		for scanner.Scan() {
+		}
+		stdout <- string(b)
+	}()
+	go func() {
+		b, _ := utilio.ReadAtMost(stderrPipe, maxReadLength)
+		scanner := bufio.NewScanner(stderrPipe)
+		for scanner.Scan() {
+		}
+		stderr <- string(b)
+	}()
+
+	if err := e.Start(); err != nil {
+		exit, ok := err.(exec.ExitError)
+		if ok && exit.ExitStatus() != 0 {
+			return probe.Failure, "", nil
+		}
+		return probe.Unknown, "", fmt.Errorf("error reading start : %s", err)
+	}
+
+	err = e.Wait()
+	data := <-stdout + <-stderr
```

Comment:
Confirming that we won't see execution hang at this point if there is either no stdout or no stderr?

Reply:
AFAICT. yes!